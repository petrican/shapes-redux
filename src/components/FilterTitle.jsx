import React from "react";

type Props = {
    shapesState: object, 
    colorsState: object
  };

const FilterTitle = (props: Props) => {
  let theTitle = props.title;

  const supLimitShapes = Object.keys(props.shapesState).length;
  const supLimitColors = Object.keys(props.colorsState).length;
  let singleColor = null;
  let singleShape = null;

  const trueColors = Object.values(props.colorsState).reduce(
    (accumulator, currentValue) => accumulator + currentValue
  );
  const trueShapes = Object.values(props.shapesState).reduce(
    (accumulator, currentValue) => accumulator + currentValue
  );

  // a. When all the colors and shapes are selected: “All items:”
  if (trueColors === supLimitColors && trueShapes === supLimitShapes) {
    theTitle = "All items";
  }

  // b. When all the colors and a multiple shapes are selected: “Multiple items:”
  if (
    trueColors === supLimitColors &&
    (trueShapes < supLimitShapes && trueShapes > 1)
  ) {
    theTitle = "Multiple items";
  }

  // c. When all the shapes and multiple colors are selected: “Multiple items:”
  if (
    trueShapes === supLimitShapes &&
    (trueColors < supLimitColors && trueColors > 1)
  ) {
    theTitle = "Multiple items";
  }

  // d. When all the shapes and a single color is selected: “All red items:”
  if (trueShapes === supLimitShapes && trueColors === 1) {
    singleColor = Object.keys(props.colorsState).filter(
      key => props.colorsState[key] === true
    );
    theTitle = `All ${singleColor} items`;
  }

  // e. When all the colors and single shape is selected: “All round items:”
  if (trueColors === supLimitColors && trueShapes === 1) {
    singleShape = Object.keys(props.shapesState).filter(
      key => props.shapesState[key] === true
    );
    theTitle = `All ${singleShape} items`;
  }

  // f. When multiple the shapes and a single color is selected: “Multiple red items:”
  if (trueShapes < supLimitShapes && trueShapes > 1 && trueColors === 1) {
    singleColor = Object.keys(props.colorsState).filter(
      key => props.colorsState[key] === true
    );
    theTitle = `Multiple ${singleColor} items`;
  }

  // g. When multiple the colors and single shape is selected: “Multiple round items:”
  if (trueColors < supLimitColors && trueColors > 1 && trueShapes === 1) {
    singleShape = Object.keys(props.shapesState).filter(
      key => props.shapesState[key] === true
    );
    theTitle = `Multiple ${singleShape} items`;
  }

  // h: When a single color and single shape is selected: “Round red items:”
  if (trueColors === 1 && trueShapes === 1) {
    singleShape = Object.keys(props.shapesState).filter(
      key => props.shapesState[key] === true
    );
    singleColor = Object.keys(props.colorsState).filter(
      key => props.colorsState[key] === true
    );
    theTitle = `${singleShape} ${singleColor} items`;
  }

  return <div className="filter-title">{theTitle}:</div>;
};

export default FilterTitle;
