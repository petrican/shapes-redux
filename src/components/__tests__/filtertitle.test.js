'use strict';

import React from 'react';
import FilterTitle from '../FilterTitle';
import renderer from 'react-test-renderer';

it('renders correctly', () => {

 const shapesState = {
        ROUND: true,
        SQUARE: true,
        TRIANGLE: true,
        OVAL: true,
        RECTANGLE: true
      };
 const colorsState = {
        grey: true,
        yellow: true,
        blue: true,
        green: true,
        red: true,
        purple: true
      };
  const tree = renderer
    .create( <FilterTitle shapesState={shapesState} colorsState={colorsState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
