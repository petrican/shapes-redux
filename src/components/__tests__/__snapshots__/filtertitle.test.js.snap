// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="filter-title"
>
  All items
  :
</div>
`;
